### ISSUE TITLE
Issue with import deepof.data related to Numba and Tensorflow

### ISSUE THREAD

**Issue by matthieuaguilera** [mlfpm/deepof#28]
Hello, I'm doing single animal tracking and try to do an unsupervised behavior classification from DLC extracted data.
I have some errors when I run import deepof.data. By ignoring this error I was still able to create a new project with a custom labelling scheme and get_graph_coord however I got another error with deep_unsupervised_embedding that might be related to the first error.
Here is the error I get with import deepof.data:

```
>>> import deepof.data
C:\Users\Recode\Anaconda3\envs\deepof\lib\site-packages\umap\distances.py:1063: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit()
C:\Users\Recode\Anaconda3\envs\deepof\lib\site-packages\umap\distances.py:1071: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit()
C:\Users\Recode\Anaconda3\envs\deepof\lib\site-packages\umap\distances.py:1086: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit()
C:\Users\Recode\Anaconda3\envs\deepof\lib\site-packages\umap\umap_.py:660: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit()
2023-08-23 10:40:04.364344: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: SSE SSE2 SSE3 SSE4.1 SSE4.2 AVX AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
C:\Users\Recode\Anaconda3\envs\deepof\lib\site-packages\ot\backend.py:2998: UserWarning: To use TensorflowBackend, you need to activate the tensorflow numpy API. You can activate it by running:
from tensorflow.python.ops.numpy_ops import np_config
np_config.enable_numpy_behavior()
  register_backend(TensorflowBackend())
```
Also when I run the 3 last command indicated I obtain this Error:
```
>>> from tensorflow.python.ops.numpy_ops import np_config
>>> np_config.enable_numpy_behavior()
>>>   register_backend(TensorflowBackend())
  File "<stdin>", line 1
    register_backend(TensorflowBackend())
IndentationError: unexpected indent
```

I was thinking in rebuilding Tensorflow as indicated but I don't know if it will affect the Tensorflow configuration used by DLC. 
Is it this that I need to do or something else?
Will it impact DLC?

Thanks in advance for your help

**Comment by lucasmiranda42** [comment]
Hi, @matthieuaguilera!

Thank you for your interest in DeepOF :)
May I ask you how you installed the package? Did you create an isolated environment?

These are all warnings that should not break DeepOF (although it would have some impact in performance). Feel free to post the error you got during unsupervised training as well, which may or may not be related :)

Best, and hope we find the problem,
Lucas

**Comment by matthieuaguilera** [comment]
Hi, thanks for your answer!
I installed the package by pip install deepof in a priorly created dedicated environment like indicated in the tutorial:

```
conda create -n deepof python=3.9
conda activate deepof
pip install deepof
```

I forgot to save the error message I got with the unsupervised training, I will rerun it to show you. However from what I remember it was a command creating NaN at each iteration and then a function in the following was not accepting NaN values.

I follow you the error message when I got it.

Thanks for your help
Best 
Matthieu
